source/common/ssl/tls_certificate_config_impl.cc

```diff
@@ -46,10 +46,6 @@ TlsCertificateConfigImpl::TlsCertificateConfigImpl(
       ocsp_staple_path_(Config::DataSource::getPath(config.ocsp_staple())
                             .value_or(ocsp_staple_.empty() ? EMPTY_STRING : INLINE_STRING)),
       private_key_method_(nullptr) {
-  if (config.has_private_key_provider() && config.has_private_key()) {
-    throw EnvoyException(fmt::format(
-        "Certificate configuration can't have both private_key and private_key_provider"));
-  }
```

Comment:
This shouldn't be deleted in this PR. This should be deleted in the later PR when we allow the provider to use this common private key.

Reply:
private key provider needs this private key to support fallback to boringssl tls handshake, so we need to remove this if.